sale_loyalty_partner_applicability/models/sale_order.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2023 Tecnativa - Pilar Vargas
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+import ast
+
+from odoo import _, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def _is_valid_partner(self, program):
+        is_valid = False
+        for rule in program.rule_ids:
+            if not is_valid and rule.rule_partners_domain != "[]":
+                domain = ast.literal_eval(rule.rule_partners_domain) + [
+                    ("id", "=", self.partner_id.id)
+                ]
+                is_valid = bool(self.env["res.partner"].search_count(domain))
+            else:
+                is_valid = True
+                break
+        return is_valid
+
+    def _program_check_compute_points(self, programs):
+        res = super()._program_check_compute_points(programs)
+        # Iterate through the programs that initially have no errors
```

Comment:
Put it as docstring and explain the why we do this, no what we're doing

Reply:
done! How do you see it? :S :P